framework/src/org/checkerframework/framework/stub/StubParser.java

```diff
@@ -845,8 +868,11 @@ private void annotateParameters(
             TypeParameter param = typeParameters.get(i);
             AnnotatedTypeVariable paramType = (AnnotatedTypeVariable) typeArguments.get(i);
 
-            annotate(paramType, param.getAnnotations());
-            if (param.getTypeBound() != null && param.getTypeBound().size() == 1) {
+            if (param.getTypeBound() == null || param.getTypeBound().size() != 1) {
```

Comment:
Is the `!= 1` case covering intersection and union types? Should it be `> 1`? Maybe add a comment why these don't need to be handled.